src/utils/createRule.js

```diff
@@ -9,7 +9,9 @@ import cloneStyle from '../utils/cloneStyle'
  */
 export default function createRule(name: string = 'unnamed', decl: JssStyle, options: RuleOptions): Rule {
   const {jss} = options
-  const declCopy = cloneStyle(decl)
+  // It has `isDynamic` property if an Observable rule was used, so no need to
+  // clone it.
+  const declCopy = decl && decl.isDynamic ? decl : cloneStyle(decl)
```

Comment:
`cloneStyle` already passes through observables.  Is it worth having this handled in both places?

Reply:
cloneStyle would skip isDynamic property which is not enumerable, we need to keep it

Reply:
mb we can remove the check from cloneStyle

Reply:
decl in this case is actually not an Observable, it is the initialStyle object we pass to createRule from the observables plugin

Reply:
Maybe fixup the comment then.

Reply:
> `isDynamic` is set in styles that are entirely specified dynamically (e.g. written to by an Observable).  In that case, we don't need to clone the style.

Reply:
Maybe I should just create that object with some `InitialStyle`constructor so that I can identify it by constructor check instead of the non-enumerable property.

Reply:
At the end it still can't be cloned  without setting again the constructor ... so not sure if its a big win.

Reply:
SGTM